[BUG]: pycaret tests failing (parallel error)

### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).


### Issue Description

some tests in pycaret failing with these messages:
FAILED tests/test_time_series_engines.py::test_engines_setup_global_args - TypeError: AutoARIMA.__init__() got an unexpected keyword argument 'parallel'
FAILED tests/test_time_series_engines.py::test_create_model_engines_local_args - TypeError: AutoARIMA.__init__() got an unexpected keyword argument 

i found a pr in sktime package saying the problem maybe related to statsforecast, see: https://github.com/sktime/sktime/commit/748106ae78c3e1c7112d88fb99acdb12c4268381

we can find a different solution or the same

### Reproducible Example

```python
.
```


### Expected Behavior

.

### Actual Results

```python-traceback
FAILED tests/test_time_series_engines.py::test_engines_setup_global_args - TypeError: AutoARIMA.__init__() got an unexpected keyword argument 'parallel'
FAILED tests/test_time_series_engines.py::test_create_model_engines_local_args - TypeError: AutoARIMA.__init__() got an unexpected keyword argument
```


### Installed Versions

<details>
github latest
</details>

--- a/requirements-optional.txt
+++ b/requirements-optional.txt
@@ -16,7 +16,8 @@ catboost>=0.23.2; platform_system != "Darwin"
 catboost>=0.23.2,<1.2; platform_system == "Darwin"  # https://github.com/pycaret/pycaret/issues/3563
 kmodes>=0.11.1
 mlxtend>=0.19.0
-statsforecast>=0.5.5
+# Can not be 1.6.0 since parallel argument in AutoARIMA was deprecated and this is not fixed in sktime
+statsforecast>=0.5.5,<1.6.0
 scikit-learn-intelex>=2023.0.1; platform_machine == 'x86_64' or platform_machine == 'AMD64'
 
 # Tuners

